SU2_CFD/include/limiters/computeLimiters_impl.hpp

```diff
@@ -0,0 +1,260 @@
+/*!
+ * \file computeLimiters_impl.hpp
+ * \brief Generic and general computation of limiters.
+ * \note Common methods are derived by defining only small details.
+ * \author P. Gomes
+ * \version 7.0.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+/*!
+ * \brief General limiter computation for methods based on one limiter
+ *        value per point (as opposed to one per edge) and per variable.
+ * \note This implementation can be used to derive most common methods
+ *       by specializing the limiter functions (e.g. Venkatakrishnan)
+ *       and the geometric modifications (e.g. sharp edges), this is done
+ *       via specialization of "CLimiterDetails" (all its methods). Then,
+ *       a call to this function with the specialized "LimiterKind" should
+ *       be added to the body of "computeLimiters()".
+ *       See also the notes in computeGradientsGreenGauss.hpp
+ *
+ * Arguments:
+ * \param[in] solver - Optional, solver associated with the field (used only for MPI).
+ * \param[in] kindMpiComm - Type of MPI communication required.
+ * \param[in] kindPeriodicComm1 - Type of periodic comm. to determine min/max.
+ * \param[in] kindPeriodicComm2 - Type of periodic comm. to adjust limiters.
+ * \param[in] geometry - Geometric grid properties.
+ * \param[in] config - Configuration of the problem.
+ * \param[in] varBegin - First variable index for which to compute limiters.
+ * \param[in] varEnd - End of computation range (nVar = end-begin).
+ * \param[in] field - Variable field.
+ * \param[in] gradient - Gradient of the field.
+ * \param[out] fieldMin - Minimum field values over direct neighbors of each point.
+ * \param[out] fieldMax - As above but maximum values.
+ * \param[out] limiter - Reconstruction limiter for the field.
+ *
+ * Template parameters:
+ * \param FieldType - Generic object with operator (iPoint,iVar)
+ * \param GradientType - Generic object with operator (iPoint,iVar,iDim)
+ * \param LimiterKind - Used to instantiate the right details class.
+ */
+template<class FieldType, class GradientType, ENUM_LIMITER LimiterKind>
+void computeLimiters_impl(CSolver* solver,
```

Comment:
And this is the general limiter implementation, the idea is that this can stay independent of the particular details of each method.
For each point we loop over its direct neighbours, as explained in #789 this is faster than looping by edges as the computation is more local (more flops, less bytes), and it is more parallelizable.